posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+The stratgy used by Open-Liberty builds is to annotate test classes and methods to indicate which classes and methods are to be run as Jakarta EE 9 tests.  Then, when running tests, if any tests are marked to be run as Jakarta EE 9 tests, the test are re-run, first performing a JavaEE to Jakarta EE 9 transformation on the test artifacts, then actually running the transformed tests.
+
+Two specific updates are required by test artifacs.  First, existing package references are updated, changing JavaEE package references to Jakarta EE 9 references, for those packages which are present in Jakarta EE 9 specifications.  Package references are in test applications, test code, and in server configurations.  Second, server configurations are updated to replace JavaEE features with the corresponding Jakarta EE 9 features.  The Eclipse Transformer may be used to perform the first of these updates.  Server configuration features must be updated manually.
+
+== Invoking the Transformer
+
+The Eclipse-Transformer can be used directly from the command line, or can be invoked as a build task, or directly from java code.  This article describes using the transformer from the command line.  See the transformer project pages for additional usage guidelines.  
+
+=== The Eclipse Transformer Project
+
+The Eclipse Transformer is an Eclipse hosted open source project:
+
+[source]
+--
+https://projects.eclipse.org/projects/technology.transformer
+--
+
+=== Downloading the Transformer 
+
+A latest snapshot of the Eclipse Transformer may be obtained, for example, as:
+
+https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/0.3.0-SNAPSHOT/org.eclipse.transformer.cli-0.3.0-20201027.133230-4-distribution.jar
+
+=== Building the transformer:
+
+Alternatively, the transformer may be build by cloning Eclipse Transformer GIT repository:
+
+https://github.com/eclipse/transformer and running:
+
+And running maven to build the transformer jar:
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area:
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unpackaging the distribution obtains:
+
+[source]
+--
+    org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar
+    libs/org.eclipse.transformer-0.3.0-SNAPSHOT.jar
+    libs/biz.aQute.bnd.transform-5.2.0.jar
+    libs/slf4j-api-1.7.25.jar
+    libs/commons-cli-1.4.jar
+    libs/slf4j-simple-1.7.30.jar
+--
+
+The jar 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' is runnable, with the invocation selecting as main class 'org.eclipse.transformer.jakarta.JakartaTransformer' and with the necessary jars from the 'libs' folder included in the class path.
+
+=== Running the transformer with default rules
```

Comment:
Instead of having a default and custom section, maybe just have `Running the transformer` and combine the two sections.

Reply:
You should still change it to `Running the transformer` or put it back to adding the `Running the transformer with custom rules.`  Right now it doesn't make sense to leave the default title, but you describe both default and custom.

Reply:
Good catch ... having collapsed the two sections the remaining title no longer makes sense.